pipelines/controller/src/main/java/org/openmrs/analytics/PipelineManager.java

```diff
@@ -243,6 +251,7 @@ synchronized void runBatchPipeline() throws IOException, PropertyVetoException,
     Preconditions.checkState(!isRunning(), "cannot start a pipeline while another one is running");
     PipelineConfig pipelineConfig = dataProperties.createBatchOptions();
     FhirEtlOptions options = pipelineConfig.getFhirEtlOptions();
+    options.setBatchRun(true);
```

Comment:
It seems we are using this new flag just for storing the pipeline mode and later accessing it, right? So for example it doesn't make sense for a direct user of the pipeline to set this. In that case, I think it makes more sense to move this to a simple class attribute in `PipelineThread`. So when you create an instance of it in `currentPipeline` below, you simply pass true/false for this flag and we don't need to leak it into `FhirEtlOptions`.

Reply:
Agreed that the batchRun flag need not be exposed via pipeline option class, refactored the code to store the value as a class variable in the PipelineThread class.